### PR TITLE
Implement `and` and `or` for `null_type`

### PR DESCRIPTION
**Pull request by raxyte**
Implements: 
- `<bool> or <null>`
- `<null> or <null>`
- `<bool> and <null>`
- `<null> and <null>`

Closes https://github.com/tenzir/issues/issues/2168

### CODE REVIEW

File: libtenzir/src/tql2/eval_binary.cpp
Comment (jachris) [inline libtenzir/src/tql2/eval_binary.cpp:1]: Please add a changelog for this. (We are now starting to do this with TQL2 changes.)

**Review by jachris — APPROVED** [review]
Thanks!

File: changelog/next/changes/4689--null-comparision-ops.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+We made [Three-Valued Logic](https://en.wikipedia.org/wiki/Three-valued_logic) comparisions possible for `null` and
+`bool` types. Additionally, these no longer emit a diagnostic when used.
```
Comment (jachris) [inline changelog/next/changes/4689--null-comparision-ops.md:2]: The three-valued logic was also there before, and typically you don't need to reference external resources in a changelog. Below is another suggestion, which also spells out the change in behavior more clearly. Finally, please stick to the maximum line length when writing changelogs.
```suggestion
The boolean operators `and`/`or` now work correctly for the type `null`.
Previously, `null and false` evaluated to `null`, and a warning was emitted.
Now, it evaluates to `false` without a warning.
```

File: changelog/next/changes/4689--null-comparision-ops.md
Comment (jachris) [inline changelog/next/changes/4689--null-comparision-ops.md:1]: Can you move this from `changes` to `bug-fixes`?

**Review by jachris — APPROVED** [review]

File: changelog/next/changes/4689--null-comparision-ops.md (hunk 1 above)
Comment (raxyte) [reply changelog/next/changes/4689--null-comparision-ops.md]: Fixed, thanks!

**Pull request merged**